src/doorman/test/TelnetProtocolTest.cpp

```diff
@@ -0,0 +1,270 @@
+#include "TelnetProtocol.hpp"
+
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+
+#include <arpa/telnet.h>
+#include <catch2/catch.hpp>
+#include <catch2/trompeloeil.hpp>
+
+namespace {
+class MockHandler : public TelnetProtocol::Handler {
+public:
+    MAKE_MOCK1(send_bytes, void(gsl::span<const byte>), override);
+    MAKE_MOCK1(on_line, void(std::string_view), override);
```

Comment:
It's not important, but vscode seems to complain about some of these macros, even after I've reconfigured it to look for the headers in ~/.conan/data.

Reply:
Aha, I think I resolved that issue.  It was also complaining about std::string_view not being a member of std:: (in this file, but not in some of your others, confusingly).   I went into the C++ extension settings and told it to use the "gnu++17" standard like so, and it looks ok now:

```    "C_Cpp.default.cppStandard": "gnu++17"
```

Reply:
Oh nice! Thanks! We should porbalby put that in the docs. And/or feel free to commit `.vscode` (which I've done in [other projects](https://github.com/compiler-explorer/infra/tree/master/.vscode)).